### PR TITLE
Add boolean flag to determine whether crossflow is allowed in a well

### PR DESCRIPTION
**Pull request by totto82**
This PR adds allow_cf to the wells structure that determine whether
crossflow is allowed or not. An extra argument is added to addWell(..)
to specify the allow_cf flag.

Should be considered together with OPM/opm-parser#600

@bska I would appreciate if you could take a look at this. I still feel shaky whenever I touching the wells in opm-core.

### CODE REVIEW

File: opm/core/wells.h [hunk 1]
```diff
@@ -106,6 +106,11 @@ struct Wells
      * Internal management structure.
      */
     void *data;
+
+    /**
+     * Array of booleans, if true crossflow is allowed in the well.
+     */
+    bool *allow_cf;
```
Comment (bska) [inline opm/core/wells.h]: Make this `int*` and change the calling interfaces accordingly.  Furthermore, move the declaration ahead of `data`.

File: opm/core/wells/WellsManager_impl.hpp [hunk 2]
```diff
@@ -247,6 +247,12 @@ void WellsManager::createWellsFromSpecs(std::vector<WellConstPtr>& wells, size_t
                     wd.type = INJECTOR;
                 else
                     wd.type = PRODUCER;
+
+                if (well->getAllowCrossFlow()) {
+                    wd.allowCrossFlow = true;
+                } else {
+                    wd.allowCrossFlow = false;
+                }
```
Comment (bska) [inline opm/core/wells/WellsManager_impl.hpp]: This is just
```C++
wd.allowCrossFlow = well->getAllowCrossFlow();
```
and should be written as such.

File: opm/core/wells/wells.c [hunk 3]
```diff
@@ -85,6 +85,7 @@ wells_allocate(int nwells, struct Wells *W)
     void *type, *depth_ref, *comp_frac;
     void *well_connpos;
     void *ctrls, *name;
+    bool *allow_cf;
```
Comment (bska) [inline opm/core/wells/wells.c]: Make this `void*`.

**Comment by atgeirr** [comment]
Looks fine, will merge.

**Comment by atgeirr** [comment]
I should of course have realized that a follow-up PR changing the add_well() calls in other modules was missing... Fixing now.

**Pull request merged**